tests/api/route-services/gem2s.test.js

```diff
@@ -54,304 +58,96 @@ const mockGem2sParamsBackendCall = (
-          samples: {
-            'sample-1': { name: 'Sample 1' },
-            'sample-2': { name: 'Sample 2' },
-          },
-        })),
-    }));
-
-    ProjectsService.mockImplementation(() => ({
-      getProject: jest.fn()
-        .mockImplementationOnce(() => Promise.resolve({ metadataKeys: [] })),
-    }));
-
-    const { taskParams: task1, hashParams: hash1 } = await Gem2sService.generateGem2sParams('experimentId');
-
-    ExperimentService.mockImplementation(() => ({
-      getExperimentData: jest.fn()
-        .mockImplementationOnce(() => Promise.resolve({
-          projectId: 'project-2', // Changed
-          experimentName: 'Another new experiment', // Changed
-          meta: {
-            organism: null,
-            type: '10x',
-          },
-          sampleIds: ['sample-1', 'sample-2'],
-        })),
-    }));
-
-    SamplesService.mockImplementation(() => ({
-      getSamplesByExperimentId: jest.fn()
-        .mockImplementationOnce(() => Promise.resolve({
-          samples: {
-            'sample-1': { name: 'Sample 1' },
-            'sample-2': { name: 'Sample 2' },
-          },
-        })),
-    }));
-
-    ProjectsService.mockImplementation(() => ({
-      getProject: jest.fn()
-        .mockImplementationOnce(() => Promise.resolve({ metadataKeys: [] })),
-    }));
-
-
-    const { taskParams: task2, hashParams: hash2 } = await Gem2sService.generateGem2sParams('experimentId');
-
-    expect(hash1).toEqual(hash2);
-  });
-
-  it('generateGem2sTaskParams - different sample order should return the same hash', async () => {
-    mockGem2sParamsBackendCall();
+    const mockedSocketsEmit = jest.fn();
+    const mockIo = {
+      sockets: {
+        emit: mockedSocketsEmit,
+      },
+    };
 
-    const { taskParams: task1, hashParams: hash1 } = await Gem2sService.generateGem2sParams('experimentId');
+    const parsedMessage = {
+      taskName: 'downloadGem',
+      experimentId: 'experimentId',
+      authJWT: 'Bearer mockAuthJwtToken',
+    };
 
-    mockGem2sParamsBackendCall({
-      sampleIds: ['sample-1', 'sample-2'], // Different sample order
-    });
+    await gem2sService.constructor.sendUpdateToSubscribed(experimentId, parsedMessage, mockIo);
 
-    const { taskParams: task2, hashParams: hash2 } = await Gem2sService.generateGem2sParams('experimentId');
+    const emitParamsChannel = mockedSocketsEmit.mock.calls[0][0];
+    expect(mockedSocketsEmit).toHaveBeenCalled();
 
-    expect(hash1).toEqual(hash2);
+    // Emitted to the correct channel
+    expect(emitParamsChannel).toMatch(experimentId);
+    expect(mockedSocketsEmit).toMatchSnapshot();
   });
 
-  it('generateGem2sTaskParams returns different hash param if passed diferent information on sample names', async () => {
-    mockGem2sParamsBackendCall();
-
-    const { taskParams: task1, hashParams: hash1 } = await Gem2sService.generateGem2sParams('experimentId');
+  it('gem2sResponse - Should return message if message is valid', async () => {
```

Comment:
Check for the case where message is invalid. Check the validity of the `GEM2SResponse.v1.yaml` schema.

Reply:
As a a follow up of this issue, a new GEM2S API schema is created (see 949b5a9). The new GEM2S API schema is still not the optimal. A ticket https://biomage.atlassian.net/browse/BIOMAGE-1460 has been opened to address this issue.

Reply:
Sounds good.